src/chunked.rs

```diff
@@ -111,36 +133,49 @@ where
         request: O::Request,
         reqs: Requirements,
     ) -> Result<ChunkedBlock<B::Memory>, MemoryError> {
+        // Check memory type
         if (1 << self.id.0) & reqs.type_mask == 0 {
             return Err(MemoryError::NoCompatibleMemoryType);
         }
-        Ok(match self.alloc_no_grow() {
-            Some(block) => {
-                assert!(block.size() >= reqs.size);
-                assert_eq!(block.range().start & (reqs.alignment - 1), 0);
-                block
-            }
+
+        // Try to allocate block
+        let block = match self.alloc_no_grow() {
+            Some(block) => block,
             None => {
+                // Grow from super-allocator
                 self.grow(owner, device, request)?;
                 self.alloc_no_grow().expect("Just growed")
             }
-        })
+        };
+
+        // Check that block meets the requirements.
+        assert!(block.size() >= reqs.size);
+        assert_eq!(block.range().start & (reqs.alignment - 1), 0);
+        Ok(block)
     }
 
     fn free(&mut self, _owner: &mut O, _device: &B::Device, block: ChunkedBlock<B::Memory>) {
-        assert_eq!(block.range().start % self.chunk_size, 0);
-        assert_eq!(block.size(), self.chunk_size);
+        assert_eq!(block.range().start % self.block_size, 0);
+        assert_eq!(block.size(), self.block_size);
         let offset = block.range().start;
         let block_memory: *const B::Memory = block.memory();
-        let block_index = unsafe {
+
+        // Dispose block retreiving chunk index
+        let chunk_index = unsafe {
             block.0.dispose();
             block.1
         };
+
+        // Confirm chunk index
         assert!(::std::ptr::eq(
-            self.blocks[block_index].memory(),
+            self.chunks[chunk_index].memory(),
             block_memory
         ));
-        let chunk_index = offset / self.chunk_size;
+
+        // Calculate block index inside chunk
+        let block_index = (offset - self.chunks[chunk_index].range().start) / self.block_size;
+
+        // Push into free blocks list
```

Comment:
not sure if omitting all the prepositions is good here, spreading "a"/"the" before used terms may make the comments more readable and add a bit of semantics